src/Symfony/Component/Notifier/Bridge/AmazonSns/Tests/AmazonSnsTransportFactoryTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\AmazonSns\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Notifier\Bridge\AmazonSns\AmazonSnsTransportFactory;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\Dsn;
+
+class AmazonSnsTransportFactoryTest extends TestCase
+{
+    public function testCreateWithDsn()
+    {
+        $factory = new AmazonSnsTransportFactory();
+        $transport = $factory->create(Dsn::fromString('sns://auth@default?region=eu-west-3&profile=myProfile'));
+        $transport->setHost('example.com');
```

Comment:
```suggestion
        $transport = $factory->create(Dsn::fromString('sns://auth@test.host?region=eu-west-3&profile=myProfile'));
```

Because we want to test the factory to use and set the custom host, not the transport, thanks.